File: js/algos/decimal2binary.js

const decimalToBinary = (decimal) => {
  let bits = ['0'];
  let rest = decimal;

  while (rest > 0) {

    if (+rest === 1) {
      bits[bits.length - 1] = '1';
      rest = 0;
    };

    for (var i = 1; Math.pow(2, i) <= decimal; i++) {
      if (decimal < Math.pow(2, i + 1)) {
        bits[i] = '1';
        for (var x = i - 1; x >= 0; x--) {
          bits[x] = '0';
        };
        rest = decimal - Math.pow(2, i);
      };
    };

    for (var y = bits.length; y >= 0; y--) {
      if (Math.pow(2, y) === decimal) {
        bits[y] = '1';
      };
    };

    decimal = rest;
  };
  return bits.reverse().join('');
};

console.log(decimalToBinary(+process.argv[2]));

// console.log((+process.argv[2]).toString(2));

module.exports = decimalToBinary;

File: py/decimal2binary.py
import sys;

def decimal2binary(decimal):
    bits = ['0'];
    rest = decimal;

    while rest > 0:


        if rest == 1:
            bits[len(bits) - 1] = '1';
            rest = 0;

        i = 1;
        while 2**i <= decimal:
            if len(bits) <= i:
                for z in range(len(bits), i + 1):
                    bits.append('0');

            if decimal < 2**(i + 1):
                bits[i] = '1';
                rest = decimal - 2**i;

                x = i;


            i += 1

        y = len(bits);
        while y >= 0:
            if 2**y == decimal:
                bits[y] = '1';
            y -= 1;

        decimal = rest;


    return ''.join(bits[::-1])

print(decimal2binary(int(sys.argv[1])))
